db/db_iter.cc

```diff
@@ -816,6 +848,17 @@ bool DBIter::FindValueForCurrentKey() {
     case kTypeValue:
       // do nothing - we've already has value in saved_value_
       break;
+    case kTypeBlobValue:
+      if (!allow_blob_) {
+        ROCKS_LOG_ERROR(logger_, "Encounter unexpected blob value.");
+        status_ = Status::NotSupported(
+            "Encounter unexpected blob value. Please open DB with "
+            "rocksdb::blob_db::BlobDB instead.");
+        valid_ = false;
+        return true;
```

Comment:
If we don't have `allow_blob` flag, what will happen? If it's not too bad, maybe we should avoid this.

Reply:
So the question is what to do if we see a blob value during scan. Here I set the iterator to invalid and return error status. What to you suggest to handle the case?